queries.js

```diff
@@ -144,11 +166,12 @@ module.exports = (self, query) => {
           const re = new RegExp(`${day}`, 'gi');
 
           query.and({
-            $and: [{ startDate: re }, { endDate: re }]
+            $and: [ { startDate: re }, { endDate: re } ]
```

Comment:
This approach doesn't catch a three day event that includes the day we're looking for in the middle. And our date prefixes and actual date fields are fully comparable as strings. So I don't get why this part was changed from the 2.x approach.

From the 2.x code:

```
      finalize: function() {
        var day = self.get('day');

        if (day === null) {
          return;
        }

        self.and({
          startDate: { $lte: day },
          endDate: { $gte: day }
        });
      },
```

We should go back to this approach for the year, month and day filters. It's simpler and more efficient than regular expressions and handles the "day in the middle" case correctly.

Reply:
I'll add this to the punch list.

Reply:
Looks like you changed your mind here, thanks for that, regressions from 2.x in terms of correctness don't make sense.

Reply:
Not sure what you mean here. That was from @bulldoguk 's original code.